app/src/features/surveys/observations/sampling-sites/components/SamplingBlockForm.tsx

```diff
@@ -0,0 +1,169 @@
+import { mdiClose, mdiMagnify } from '@mdi/js';
+import Icon from '@mdi/react';
+import { Card, CardHeader, Collapse, IconButton, Typography } from '@mui/material';
+import Autocomplete, { createFilterOptions } from '@mui/material/Autocomplete';
+import Box from '@mui/material/Box';
+import { grey } from '@mui/material/colors';
+import TextField from '@mui/material/TextField';
+import { SurveyContext } from 'contexts/surveyContext';
+import { useFormikContext } from 'formik';
+import { IGetSampleBlockRecord, IGetSurveyBlock } from 'interfaces/useSurveyApi.interface';
+import { default as React, useContext, useState } from 'react';
+import { TransitionGroup } from 'react-transition-group';
+import { ICreateSamplingSiteRequest } from '../SamplingSitePage';
+
+export const SamplingBlockForm = () => {
+  const { setFieldValue } = useFormikContext<ICreateSamplingSiteRequest>();
+  const surveyContext = useContext(SurveyContext);
+
+  const options = surveyContext.surveyDataLoader?.data?.surveyData?.blocks || [];
+
+  const [selectedBlocks, setSelectedBlocks] = useState<IGetSurveyBlock[]>([]);
+
+  interface IBlockCard {
+    label: string;
+    description: string;
+  }
+
+  const BlockCard: React.FC<IBlockCard> = (props) => (
+    <Box>
+      <Box>
+        <Typography variant="subtitle1" fontWeight="bold">
+          {props.label}
+        </Typography>
+      </Box>
+      <Box my={0.25}>
+        <Typography variant="subtitle2" color="textSecondary">
+          {props.description}
+        </Typography>
+      </Box>
+    </Box>
+  );
+
+  const [searchText, setSearchText] = useState('');
+
+  const handleAddBlock = (block: IGetSurveyBlock) => {
+    selectedBlocks.push(block);
```

Comment:
`selectedBlocks` is immutable. (See my other comment on this)

Reply:
Fixed